src/charm.py

```diff
@@ -295,18 +296,15 @@ def _on_openfga_store_removed(self, event: OpenFGAStoreRemovedEvent) -> None:
 
         self._holistic_handler(event)
 
-    def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
-        if not container_connectivity(self):
-            event.defer()
-
-        self._workload_service.push_ca_certs(event.ca)
-        self._holistic_handler(event)
-
-    def _on_certificate_removed(self, event: CertificateRemovedEvent) -> None:
+    def _on_certificate_changed(
+        self, event: CertificateAvailableEvent | CertificateRemovedEvent
+    ) -> None:
         if not container_connectivity(self):
             event.defer()
+            return
 
-        self._workload_service.remove_ca_certs()
+        certs = TLSCertificates.load(self.certificate_transfer_requirer)
+        self._workload_service.push_ca_certs(certs.ca_bundle)
```

Comment:
The problem with this is that in case of restart (or update) the function will not run, but the pod will be restarted which means that all the certs will be gone.

Reply:
OK, so probably it's better to move it to the holistic method.